src/code/z_lifemeter.c

```diff
@@ -1,17 +1,446 @@
+                                    (s32)((posX + halfTexSize) * 4), (s32)((posY + halfTexSize) * 4), G_TX_RENDERTILE,
+                                    0, 0, (s32)temp_f4, (s32)temp_f4);
+            }
+        } else {
+            if ((ddCount < 0) || (ddCount < i)) {
+                if (curCombineModeSet != 2) {
+                    curCombineModeSet = 2;
+                    func_8012C8D4(gfxCtx);
+                    gDPSetCombineLERP(OVERLAY_DISP++, PRIMITIVE, ENVIRONMENT, TEXEL0, ENVIRONMENT, TEXEL0, 0, PRIMITIVE,
+                                      0, PRIMITIVE, ENVIRONMENT, TEXEL0, ENVIRONMENT, TEXEL0, 0, PRIMITIVE, 0);
+                    gDPSetAlphaCompare(OVERLAY_DISP++, G_AC_THRESHOLD);
+                }
+            } else {
+                if (curCombineModeSet != 4) {
+                    curCombineModeSet = 4;
+                    func_8012C8D4(gfxCtx);
+                    gDPSetCombineLERP(OVERLAY_DISP++, ENVIRONMENT, PRIMITIVE, TEXEL0, PRIMITIVE, TEXEL0, 0, PRIMITIVE,
+                                      0, ENVIRONMENT, PRIMITIVE, TEXEL0, PRIMITIVE, TEXEL0, 0, PRIMITIVE, 0);
+                    gDPSetAlphaCompare(OVERLAY_DISP++, G_AC_THRESHOLD);
+                }
+            }
+            {
+                Mtx* temp_a0 = GRAPH_ALLOC(gfxCtx, sizeof(Mtx));
+                func_801780F0(temp_a0, 1.0f - (0.32f * lifesize), 1.0f - (0.32f * lifesize), 1.0f - (0.32f * lifesize),
+                              -130.0f + offsetX, 94.5f - offsetY, 0.0f);
+                gSPMatrix(OVERLAY_DISP++, temp_a0, G_MTX_LOAD | G_MTX_MODELVIEW);
+                gSPVertex(OVERLAY_DISP++, beatingHeartVtx, 4, 0);
+                gSP1Quadrangle(OVERLAY_DISP++, 0, 2, 3, 1, 0);
+            }
+        }
+        offsetX += 10.0f;
+        if (i == 9) {
+            offsetY += 10.0f;
+            offsetX = 0.0f;
+        }
+    }
+    CLOSE_DISPS(gfxCtx);
+}
+
+void LifeMeter_UpdateSizeAndBeep(GlobalContext* globalCtx) {
+    InterfaceContext* interfaceCtx = &globalCtx->interfaceCtx;
+    if (interfaceCtx->lifeSizeChangeDirection != 0) {
+        interfaceCtx->lifeSizeChange--;
+        if (interfaceCtx->lifeSizeChange <= 0) {
+            interfaceCtx->lifeSizeChange = 0;
+            interfaceCtx->lifeSizeChangeDirection = 0;
+            if (func_801233E4(globalCtx) == 0 && (globalCtx->pauseCtx.state == 0) &&
+                (globalCtx->pauseCtx.debugState == 0) && LifeMeter_IsCritical() && func_801690CC(globalCtx) == 0) {
+                play_sound(NA_SE_SY_HITPOINT_ALARM);
+            }
+        }
+    } else {
+        interfaceCtx->lifeSizeChange += 1;
+        if ((s32)interfaceCtx->lifeSizeChange >= 0xA) {
+            interfaceCtx->lifeSizeChange = 0xA;
+            interfaceCtx->lifeSizeChangeDirection = 1;
+        }
+    }
+}
+
+bool LifeMeter_IsCritical(void) {
+    s16 criticalThreshold;
+    if (gSaveContext.healthCapacity <= 0x50) {
```

Comment:
```suggestion
    s16 criticalThreshold;
    
    if (gSaveContext.healthCapacity <= 0x50) {
```
I would include everything in this function as numbers that need to be switched to decimal

Reply:
My thought was it's easier to understand what it checks here in hexa ( because 1 heart is 0x10) but I'll add it as a comment and change it in decimal if you think it's more fitting.

Reply:
Health usually is used in decimal. So, yeah, could you change it?